picasso/src/main/java/com/squareup/picasso3/BitmapUtils.kt

```diff
@@ -186,6 +189,46 @@ internal object BitmapUtils {
     }
   }
 
+  /**
+   * Decode a byte stream into a Drawable. This method will take into account additional information
+   * about the supplied request in order to do the decoding efficiently.
+   */
+  fun decodeDrawableStream(source: Source, request: Request): Drawable {
+    val exceptionCatchingSource = ExceptionCatchingSource(source)
+    val bufferedSource = exceptionCatchingSource.buffer()
+    val drawable =
+      if (VERSION.SDK_INT >= 28) {
+        val imageSource = ImageDecoder.createSource(ByteBuffer.wrap(bufferedSource.readByteArray()))
```

Comment:
I feel like in POS, we like to extract many things out to extension functions to improve readability, but maybe this would be an overkill here?

```kotlin
private fun BufferedSource.toImageDecoder(): ImageDecoder.Source =
  ImageDecoder.createSource(wrap(readByteArray()))
```

Reply:
With that line reduced the `if` statement might be able to be pulled up next to the `drawable` declaration?

Reply:
I think the extension function is overkill but am able to pull the if up a line which does help a bit